src/v2/pdf_resources/page_cell.h

```diff
@@ -262,10 +262,38 @@ namespace pdflib
   
   bool pdf_resource<PAGE_CELL>::is_adjacent_to(pdf_resource<PAGE_CELL>& other, double eps)
   {
-    double d0 = std::sqrt((r_x1-other.r_x0)*(r_x1-other.r_x0) + (r_y1-other.r_y0)*(r_y1-other.r_y0));
-    double d1 = std::sqrt((r_x2-other.r_x3)*(r_x2-other.r_x3) + (r_y2-other.r_y3)*(r_y2-other.r_y3));
+    // This assumes (even for right-to-left text) that other is to the
+    // right of this.  If two cells overlap then they are obviously
+    // adjacent, otherwise the right side corners of this must be
+    // within eps of the left side corners of other.
+
+    // Intersection of bounding rectangles (FIXME: Does not actually
+    // imply overlap for rotated cells, not immediately sure the
+    // correct and efficient way to compute that)
```

Comment:
Actually, if we are going to update this code, let's think this through how we deal with rotated cells.

Reply:
Probably a good idea - is there documentation on what the rotated coordinates (`r_x0`, `r_y0`, etc, etc) mean exactly?  I assume that you take the four corners of the bounding box in user space and apply the CTM?  Are they normalized in any way (i.e. so `r_x0, r_y0` is always the bottom left point)?

Reply:
In any case it's just the intersection of two convex polygons - and you don't actually need to find the resulting polygon, just its area (and even then you just need to know if the area is non-zero)

Reply:
Presumably we can just use the algorithm from `shapely.intersects`: https://shapely.readthedocs.io/en/2.0.6/reference/shapely.intersects.html

Reply:
Ah, that algorithm is actually in GEOS: https://github.com/libgeos/geos/blob/f1519c182497a99db8315ef78e0ae283b0469008/src/geom/Geometry.cpp#L291

Reply:
likely it can be done by testing:
- if any corner point of one quadrilateral is inside the other
- if any line segment in one quadrilateral crosses the other

Reply:
@dhdaines Yes, currently the convention for the coordinates is,

```
      rx_4, ry_4        rx_3, ry_3
         +----------------+
         |                |
         |                |
         |                |
         +----------------+
      rx_0, ry_0        rx_1, ry_1
```
      
Let me look into the algos you send the links for. I do think we need to get this right, there are other example pdf's where we have similar problems. 
  
However, I do not want to break this example: [page](https://github.com/DS4SD/docling-parse/blob/main/docs/visualisations/table_of_contents_01.pdf.page_1.char.png)

Reply:
thank you! I understand the function of `rx_*, ry_*` now.  Are `x0, y0, x1, y1` the bounding box of this rotated rectangle, or are they just the opposite corners of the user-space bounding box, after translation?  (if the latter, you may wish to consider this problem: https://github.com/pdfminer/pdfminer.six/issues/1067)

Reply:
You may wish to use the approach I use in PLAYA to verify the behaviour with various possible types of transformation: https://github.com/dhdaines/playa/blob/ad15b4b4eefc529ebd6e079fcc8411e37466c911/tests/test_lazy_api.py#L104

Reply:
yes, so consider the (rx_*, ry_*) as coordinates of a polygon. (rx_0, ry_0) is the left bottom starting point of the writing and allows you to see the orientation of writing. The x/y coordinates are defined as 

x0 = min(rx_0, rx_1, rx_2, rx_3)
y0 = min(ry_0, ry_1, ry_2, ry_3)
x1 = max(rx_0, rx_1, rx_2, rx_3)
y1 = max(ry_0, ry_1, ry_2, ry_3)

Hence, they can not provide you the orientation of writing. 

Notice also that the coordinate origin is always (in pdf) bottom left corner of the page-image (as opposed to top left often the case with image libraries).

If we are going to solve this problem, we should not use the x0, y0, x1, y1 coordinates, but only the r*_* coordinates.

Reply:
@dhdaines could you let us know if you can address some of the open points raised in this discussion? I would love if this proceeds but I cannot dig deeper into the specifics at this time. Many thanks for your help!

Reply:
> @dhdaines could you let us know if you can address some of the open points raised in this discussion? I would love if this proceeds but I cannot dig deeper into the specifics at this time. Many thanks for your help!

yes absolutely!  I haven't had time in the last 2 weeks but I'll take a look at it today.  It may be the case that we can just fix the underlying issue with overlapping glyph bboxes - possibly the charspace or wordspace are being applied incorrectly in the case where each glyph is a separate text cell.

Reply:
See updates in #99 - the underlying problem is that `docling-parse` disregards the coordinate space in Type3 fonts.  So this whole MR is mostly irrelevant, if we can fix that problem (which should be A Simple Matter Of Matrix Multiplication™)